Sources/JobsRedisDriver/JobsRedisDriver.swift

```diff
@@ -30,53 +30,43 @@ extension JobsRedisDriver: JobsPersistenceLayer {
     public func get(key: String) -> EventLoopFuture<JobStorage?> {
         let processing = processingKey(key: key)
         
-        return database.newConnection(on: eventLoop).flatMap { conn in
-            return conn.rpoplpush(source: key, destination: processing).and(result: conn)
-        }.flatMap(to: (RedisData, RedisClient).self) { redisData, conn in
+        return client.rpoplpush(from: key, to: processing).flatMap { redisData -> EventLoopFuture<String?> in
             guard let id = redisData.string else {
-                conn.close()
-                throw Abort(.internalServerError)
+                return self.eventLoop.makeFailedFuture(Abort(.internalServerError))
             }
             
-            return conn.rawGet(id).and(result: conn)
-        }.map { redisData, conn in
-            conn.close()
-            
-            guard let data = redisData.data else {
+            return self.client.get(id)
+        }.flatMapThrowing { redisData in
+            guard let data = redisData?.data(using: .utf8) else {
                 print("Could not convert redis data to Data")
                 return nil
             }
             
             let decoder = try JSONDecoder().decode(DecoderUnwrapper.self, from: data)
             return try JobStorage(from: decoder.decoder)
-        }.catchMap { error in
-            return nil
+        }.flatMapError { _ in
+            return self.eventLoop.makeSucceededFuture(nil)
         }
     }
     
     /// See `JobsPersistenceLayer.set`
     public func set(key: String, jobStorage: JobStorage) -> EventLoopFuture<Void> {
-        return database.newConnection(on: eventLoop).flatMap(to: (RedisData, RedisClient).self) { conn in
-            let data = try JSONEncoder().encode(jobStorage).convertToRedisData()
-            return conn.set(jobStorage.id, to: data).transform(to: (data, conn))
-        }.flatMap { data, conn in
-            return conn.lpush([try jobStorage.id.convertToRedisData()], into: key).transform(to: conn)
-        }.map { conn in
-            return conn.close()
+        guard let data = try? JSONEncoder().encode(jobStorage).convertedToRESPValue() else {
```

Comment:
Might be nice to not throw this error away with `try?`. A `do/catch` statement would let us forward the underlying error including any information about which keys/values are causing problems.